Add documentation for CaseDescription (#130)
ESSS/alfasim-sdk: docs/source/alfacase/case_description.quick_start.rst

Quick Start
===========

In this section, it is shown how to work with a ``CaseDescription`` and how to convert an :program:`ALFAcase`
from an existent project (exported from ALFAsim) to a ``Description``.

CaseDescription
---------------

The ``CaseDescription`` is a Python class that helps users to create a valid ``.alfacase`` file programmatically, by instantiating a
:class:`CaseDescription <alfasim_sdk.CaseDescription>` object.

> To use create a ``CaseDescription`` instance it is necessary to have a Python Interpreter with at least version 3.6 and the

Reviewer: ```suggestion
To create a ``CaseDescription`` instance it is necessary to have a Python Interpreter with at least version 3.6 and the
```

|sdk| Python package.

.. note::

    For more details on how to install Python `check the official docs <https://www.python.org/downloads/>`_.
    It is recommended that you install the |sdk| using a Python Virtual Environment.
    For more details, see `Virtual Environments and Packages in Python documentation <https://docs.python.org/3/tutorial/venv.html>`_.

From a terminal, and inside a virtual environment, update pip:

>>> python -m pip install -U pip

Install the |sdk| with:

>>> pip install alfasim-sdk

.. note::

    All classes and constants that will be used to configure a project can be accessed through the main module of ``alfasim_sdk``

> In section :ref:`alfacase-reference-section` it is possible to check all available classes and options,
> as well which types and validators are applied at the time of object instantiation.

Reviewer: ```suggestion
See section :ref:`alfacase-reference-section` for all available classes and options,
as well as which types and validators are applied at the time of object instantiation.
```

> To know the type is important to create instantiate a class properly, for example, if you try to create a
> :class:`NodeDescription <alfasim_sdk.NodeDescription>`, the ``node_type`` attribute
> must be a string one of the options available on :class:`NodeCellType <alfasim_sdk.NodeCellType>`, otherwise, a
> ``ValueError`` is raised.

Reviewer: ```suggestion
Knowing the type is important to instantiate a class properly. For example, if you try to create a
:class:`NodeDescription <alfasim_sdk.NodeDescription>`, the ``node_type`` attribute
must be a string of one of the options available on :class:`NodeCellType <alfasim_sdk.NodeCellType>`, otherwise, a
``ValueError`` is raised.
```

.. code-block:: python

    >>> import alfasim_sdk

>     >>> alfasim_sdk.NodeDescription(name=1, node_type=alfasim_sdk.NodeCellType.Pressure)

Reviewer: You could complement the example with a valid instantiation afterwards.

    [ ... ]
    ValueError: 'node_type' must be in <enum 'NodeCellType'> (got 1)

As already informed at the beginning of the section, ALFAsim accepts only ``.alfacase`` files, the description classes
are only tools that help the creation and manipulation of a project programmatically in an easier way.

> To convert a ``Description`` to alfacase you can use the function ``convert_description_to_alfacase`` and to convert a

Reviewer: I just noticed the names of the functions are not equivalent, one says `description` and the other `case`. I'm too much familiar with referencing CaseDescription as both case and description, but maybe it is confusing for users?

Author: Which one we should stick with?
After writing these docs I kind of liked to refer it as `Descriptions`

Reviewer: Yep, _case_ may look like it is referring to the alfacase files.

``.alfacase`` file to ``Description`` you can use the function ``convert_alfacase_to_case``.

Convert an ALFAsim Project to a Description
-------------------------------------------

One of the easiest ways to create a ``.alfacase`` file is converting an existent project from an ALFAsim application.

> It's also possible to convert this ``.alfacase`` file to a ``Description`` class using the function ``convert_alfacase_to_case``.

Reviewer 2: ```suggestion
It is also possible to convert this ``.alfacase`` file to a ``Description`` class using the function ``convert_alfacase_to_case``.
```

Avoid contractions in formal text.

>>> pip install alfasim-sdk

.. note::

    All classes and constants that will be used to configure a project can be accessed through the main module of ``alfasim_sdk``

See section :ref:`alfacase-reference-section` for all available classes and options,
as well as which types and validators are applied at the time of object instantiation.

Knowing the type is important to create instantiate a class properly. For example, if you try to create a
:class:`NodeDescription <alfasim_sdk.NodeDescription>`, the ``node_type`` attribute
must be a string of one of the options available on :class:`NodeCellType <alfasim_sdk.NodeCellType>`, otherwise, a
``ValueError`` is raised.

.. code-block:: python

    >>> import alfasim_sdk
    >>> alfasim_sdk.NodeDescription(name="Node", node_type=1)
    [ ... ]
    ValueError: 'node_type' must be in <enum 'NodeCellType'> (got 1)

    >>> alfasim_sdk.NodeDescription(name='Node', node_type=alfasim_sdk.NodeCellType.Pressure)

As already informed at the beginning of the section, ALFAsim accepts only ``.alfacase`` files, the description classes
are only tools that help the creation and manipulation of a project programmatically in an easier way.

> To convert a ``Description`` to alfacase you can use the function ``convert_description_to_alfacase`` and to convert a

Reviewer: I was actually talking about the functions, they are not consistent with each other, do you think it is worth renaming it?

Author: Sure, are your thoughts on this @nicoddemus @prusse-martin 

We will keep

`convert_description_to_alfacase`

And change

`convert_alfacase_to_case` to `convert_alfacase_to_description`

we can keep a function with the old name as backward compatibility at `alfasim_sdk.__init__.py` but IMHO we should remove it immediately =)

Reviewer 2: 👍 to `convert_description_to_alfacase` and `convert_alfacase_to_description`.

Agree that we don't need to keep backward compatibility.
